app/lib/actions/pulp/repository/sync.rb

```diff
@@ -114,6 +101,11 @@ def ignored_tags
           # ignore background download tasks
           ["pulp:action:download"]
         end
+
+        def smart_proxy(capsule_id = nil)
+          smart_proxy = capsule_id ? SmartProxy.unscoped.find(input[:capsule_id]) : SmartProxy.first
+          smart_proxy
+        end
```

Comment:
mind moving this to 'actions/pulp/abstract.rb' ?

I think you also need to check input[:capsule] here on line 106, not 'capsule_id'.   Also, i think you want to default to SmartProxy.default_capsule, not .first